components/QrCodeOverlay.tsx

```diff
@@ -35,6 +37,34 @@ export const QrCodeOverlay: React.FC<QrCodeOverlayProps> = props => {
     return qrData;
   }
 
+  const [base64String, setBase64String] = useState('');
+
+  let qrRef = useRef(null);
+
+  function updateBase64() {
+    qrRef.current.toDataURL(dataURL => {
+      setBase64String(`data:image/png;base64,${dataURL}`);
+    });
+    shareQRCode();
+  }
+
+  useEffect(() => {
+    shareQRCode();
+  }, [base64String]);
```

Comment:
@balachandarg-tw @KiruthikaJeyashankar why we need shareQRCode in useEffect. Share QRCode is manual step by end user.